From: =?utf-8?q?Rog=C3=A9rio_Brito?= <rbrito@gmail.com>
Date: Fri, 12 Mar 2021 05:32:22 -0300
Subject: pdfsizeopt: Use standard python2 shebang.

---
 pdfsizeopt | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/pdfsizeopt b/pdfsizeopt
index 6303162..e3c019e 100755
--- a/pdfsizeopt
+++ b/pdfsizeopt
@@ -1,22 +1,4 @@
-#!/bin/sh
-
-""":" # pdfsizeopt: PDF file size optimizer
-
-P="$(readlink "$0" 2>/dev/null)"
-test "$P" && test "${P#/}" = "$P" && P="${0%/*}/$P"
-test "$P" || P="$0"
-Q="${P%/*}"/pdfsizeopt_libexec/python
-test -f "$Q" && exec "$Q" -E -- "$P" ${1+"$@"}
-type -p python2.7 >/dev/null 2>&1 && exec python2.7 -- "$P" ${1+"$@"}
-type -p python2.6 >/dev/null 2>&1 && exec python2.6 -- "$P" ${1+"$@"}
-type -p python2.5 >/dev/null 2>&1 && exec python2.5 -- "$P" ${1+"$@"}
-type -p python2.4 >/dev/null 2>&1 && exec python2.4 -- "$P" ${1+"$@"}
-exec python -- "$P" ${1+"$@"}; exit 1
-
-This is a Python 2.x script, it works with Python 2.4, 2.5, 2.6 and 2.7. It
-doesn't work with Python 3.x. Feel free to replace the #! line with
-`#! /usr/bin/python', `#! /usr/bin/env python' or whatever suits you best.
-"""
+#!/usr/bin/python2
 
 import os
 import os.path
